### PR TITLE
Add support for vorbis covers.

### PR DESCRIPTION
**Pull request by smimram**

### CODE REVIEW

**Comment by toots** [comment]
I think you might to have a look at: https://github.com/savonet/ocaml-ogg/blob/master/src/ogg_demuxer.mli

**Comment by toots** [comment]
Also, we have a generic path to do that, via `request.resolve`/`request.metadata`. Is there a need for a vorbis/ogg specific tool?

**Comment by smimram** [comment]
I'll have a look at the ogg demuxer. The reason why I want something custom is that usual tag readers such as taglib don't export cover art, which is what I am interested in right now.

**Comment by smimram** [comment]
You were right, I can actually use the standard metadata for ogg!

**Comment by smimram** [comment]
Are you ok now?

File: CHANGES.md [hunk 1]
```diff
@@ -59,6 +58,8 @@ Changed:
 - Renamed `which` to `file.which`.
 - Change `blank()` duration semantics to mean forever only on negative values.
 - Get rid of numbering of universal variables (#1037).
+- Renamed `base64.decode`/`base64.encode` to
+  `string.base64.decode`/`string.base64.encode`.
```
Comment (toots) [inline CHANGES.md]: 👍

File: libs/metadata.liq [hunk 2]
```diff
@@ -33,17 +33,50 @@ def metadata.float_getter(init, metadata, s)
...
 def file.cover(file)
-  m = file.mp3.tags(file)
-  m = list.filter(fun (lv) -> fst(lv) == "APIC", m)
-  m = list.map(snd, m)
-  if m == [] then "" else
-    # TODO: we could use file type in order to select cover if there are many
-    let (_,_,_,data) = file.mp3.parse_apic(list.hd(default="", m))
-    data
+  m = file.metadata(file)
+  print("**meta: #{m}")
```
Comment (toots) [inline libs/metadata.liq]: Remove?
```suggestion
```

File: libs/metadata.liq [hunk 3]
```diff
@@ -33,17 +33,50 @@ def metadata.float_getter(init, metadata, s)
...
+  m = list.map (fun (lv) -> (string.case(lower=true, fst(lv)), snd(lv)), m)
+  if list.exists(fun (lv) -> fst(lv) == "coverart", m) then
```
Comment (toots) [inline libs/metadata.liq]:
```suggestion
  if list.mem_assoc("coverart", m) then
```

File: libs/metadata.liq [hunk 4]
```diff
@@ -33,17 +33,50 @@ def metadata.float_getter(init, metadata, s)
...
+    # We are likely to have an ogg/vorbis file
+    m = list.filter(fun (lv) -> fst(lv) == "coverart", m)
+    m = list.map(snd, m)
+    m = list.hd(default="", m)
```
Comment (toots) [inline libs/metadata.liq]:
```suggestion
    m = list.assoc(default="", "coverart", m)
```
And remove the two lines above.

File: libs/metadata.liq [hunk 5]
```diff
@@ -33,17 +33,50 @@ def metadata.float_getter(init, metadata, s)
...
+    if m == "" then "" else string.base64.decode(m) end
+  else
+    # Assume we have an mp3 file
+    m =
+      if list.exists(fun (lv) -> fst(lv) == "apic", m) then
```
Comment (toots) [inline libs/metadata.liq]:
```suggestion
      if list.mem_assoc("apic", m) then
```

File: libs/metadata.liq [hunk 6]
```diff
@@ -33,17 +33,50 @@ def metadata.float_getter(init, metadata, s)
...
+      else
+        # Try the builting tag reader because APIC tags are not read by default,
+        # e.g. with taglib
+        file.mp3.metadata(file)
+      end
+    m = list.filter(fun (lv) -> fst(lv) == "apic", m)
+    m = list.map(snd, m)
+    if m == [] then
+      ""
+    else
```
Comment (toots) [inline libs/metadata.liq]:
```ruby
m = list.assoc(default="", "apic", m)
if m == "" then
```
And update below..

**Pull request merged**